### ISSUE TITLE
Handle white space better

### ISSUE THREAD

**Issue by pseudotensor** [h2oai/datatable#514]
[OrdonezA_ADLs.txt](https://github.com/h2oai/datatable/files/1412743/OrdonezA_ADLs.txt)

**Comment by pradkrish** [comment]
@st-pasha 

To fix this bug, should repeated tabs be treated as a single tab? For example, the following row
```python
14.11	14.1	0.8911	5.42	3.302	2.7		5		1
```
is internally treated as 
```python
14.11\t14.1\t0.8911\t5.42\t3.302\t2.7\t\t5\t\t1\n
```
and ends up padding extra columns

**Comment by st-pasha** [comment]
Well, the problem is that TSV (tab-separated values) is a popular format where the values are separated with tabs. So if you have a row like `[1, NA, 2]` then it ends up encoded as `1\t\t2\n`, which is why tabs must be treated as separate delimiters.

The *correct* solution here is first to try parsing the file as a table of values, where each column is expected to be vertically aligned with arbitrary number of tabs/spaces. We can try different choices of tab widths too: 8, 4, 2.
Note that even within a table there could be gaps where the values are missing, in which case it will look like a sequence of whitespace spanning multiple columns.

**Comment by pradkrish** [comment]
is that so, the example you chose `[1, NA, 2]` is encoded as `1\tNA\t2`.  `[1, , 2]` is encoded as `1\t\t2\n`.

How about this: let's say, in the first run, we remove all the extra tabs (which are placeholders for missing values) and check the number of inferred columns of each row. If they all match, then that is the most likeliest configuration. If they don't match, we just reintroduce the removed tabs from before and carry on as usual. Let me illustrate this with a `3*3` example

[1,2,NA]                                                                     
[1,   ,2]  -------->  `1\t2\tNA\n1\t\t2\n1\t\t2\n` will become `1\t2\tNA\n1\t2\n1\t2\n` 
[1,   ,2]                                                    
    
After removing extra tabs, `row1` will be 3 columns, `row2` will be 2 columns, `row3` will be 2 columns. Since they don't match, reintroduce the removed tabs and run again, the result will be a `(3 * 3)` table. As a second example

[1,2,  ]                                                                     
[1,   ,2]  -------->  `1\t2\n1\t\t2\n1\t\t2\n` will become `1\t2\n1\t2\n1\t2\n` 
[1,   ,2] 
                                                
After removing extra tabs, `row1` will be 2 columns, `row2` will be 2 columns, `row3` will be 2 columns. Since they do match, that is the likeliest configuration and the result will be a `(3 * 2 )` table. This is in essence what's happening in [another.zip](https://github.com/h2oai/datatable/files/1423962/another.zip). Here, I am assuming that we would like to treat the second example as a `3*2` table and not as a `3*3` table, which is why someone had raised this issue in the first place. 

According to this logic, we run through every frame twice. Let me know what you think.

**Comment by st-pasha** [comment]
So what you are suggesting is that we have a new separator: `\t+`, which means "one or more tabs", similar to how we already have separator `\s+`. And if during the detection phase it manages to produce a more consistent number of columns than `\t` separator, then we should use `\t+` for parsing.

I agree that this would parse `another.zip`, and the original Ordonez file (assuming we trim the trailing tabs).

However, I'm worried that this would not work if the data contained any gaps (NAs). Whenever there is any such gap, two separators will be next to each other, and will be processed as a single separator, causing all subsequent fields to be shifted.
For example, consider the following example:
```
Start time    End time     Activity
02:27:59      10:18:11     Sleeping
10:21:24                   Toileting
10:25:44      10:33:00     Showering
```
Even without counting the spaces or tabs here, you know just by looking that there are 3 columns here, and that the 2nd row should be read as `("10:21:24", None, "Toileting")`. And the reason you know this is that the columns are visually aligned to the same positions. So, a reader ultimately has to implement the same logic: detect if the input consists of vertically aligned columns, and then parse accordingly (which won't even be a csv reader).

**Comment by pradkrish** [comment]
If we were to remove all the extra tabs for missing values in your example, `row0,` `row1` and `row3` will be inferred to contain 3 columns while `row2` will have 2 columns. Since the values don't match, according to my logic, we will fall back to the current way of arranging. It's only when the number of inferred columns for all rows match, we go with the new method. I don't see why it doesn't work.

**Comment by st-pasha** [comment]
Yes, but the current way of arranging will also fail because there are multiple tabs/spaces between the values there. So the dataset will fail to parse, even though it looks perfectly "normal" to human eye.
